project/cmake/scripts/common/projectmacros.cmake

```diff
@@ -50,10 +50,32 @@ endfunction()
 # Copied from FindGTest.cmake 
 # Thanks to Daniel Blezek <blezek@gmail.com> for the GTEST_ADD_TESTS code
 function(GTEST_ADD_TESTS executable extra_args)
+    get_property(XBMC_GENERATED_TEST_FILES GLOBAL PROPERTY XBMC_GENERATED_TEST_FILES)
```

Comment:
yeah. that's the one i was referring to. but if you read my other comment, i think that suggestion is a better one. then we don't have to hack around in gtest_add_tests at all. if we distinguish between sources going into the test .a and things added to test_SOURCES in core_add_test, we can avoid this completely.